server/container/util-java8/src/main/java/org/apache/james/util/FluentFutureStream.java

```diff
@@ -69,11 +80,28 @@ private FluentFutureStream(CompletableFuture<Stream<T>> completableFuture) {
             CompletableFutureUtil.thenComposeOnAll(completableFuture(), function));
     }
 
+    public <U> FluentFutureStream<U> thenFlatCompose(Function<T, CompletableFuture<Stream<U>>> function) {
```

Comment:
You can hardly do that without accord from the community...

Let's first use java-8 everywhere....